src/preset_cli/cli/superset/sync/dbt/datasets.py

```diff
@@ -109,9 +112,7 @@ def sync_datasets(  # pylint: disable=too-many-locals, too-many-branches, too-ma
 
         dataset_metrics = []
         model_metrics = {
-            metric["name"]: metric
-            for metric in metrics
-            if model["unique_id"] in metric["depends_on"]
```

Comment:
This was too naive, and didn't account for how derived metrics **depend on other metrics**, instead of depending on models.